src/document.c

```diff
@@ -507,10 +507,16 @@ FIELD_BULK_INDEXER(numericIndexer) {
 }
 
 FIELD_PREPROCESSOR(vectorPreprocessor) {
-  size_t len;
-  fdata->vector = RedisModule_StringPtrLen(field->text, &len);
-  fdata->vecLen = len;
-  if (len != fs->vectorOpts.expBlobSize) {
+  fdata->vecLen = 0;
+  if (field->unionType == FLD_VAR_T_RMS) {
+    fdata->vector = RedisModule_StringPtrLen(field->text, &fdata->vecLen);
+  } else if (field->unionType == FLD_VAR_T_CSTR) {
+    fdata->vector = field->strval;
+    fdata->vecLen = field->strlen;
+  } else if (field->unionType == FLD_VAR_T_NULL) {
+    return 0; // Skipping indexing missing vector
```

Comment:
Should fail?

Reply:
Why? isn't all indexes return 0 and later skips indexing?